### DISCUSSION TITLE
Dynamic API Routes fail in production build

### DISCUSSION THREAD

**Discussion by jguipi** [vercel/next.js#17552] [Help] [answered]
# Bug report

## Describe the bug
Dynamic ~~API~~ routes `(pages/test/[index.js])` fails top build when I try to deploy into production builds (yarn build followed by yarn start). It returns an Internal Server Error with the following stack:
```
> Build error occurred
Error: Build optimization failed: found page without a React Component as default export in pages/test
```
https://nextjs.org/docs/routing/dynamic-routes

## Expected behaviour

The build should be successful

## System information

- OS: macOS
- Browser: chrome
- Version of Next.js: 9.5.3
- Version of Node.js: v10.16.3

**Comment by ijjk** [comment]
Hi, API routes must be nested under `pages/api` all other pages outside of this directory are expected to export a react component as the default export. See more info on API routes in the documentation [here](https://nextjs.org/docs/api-routes/introduction).

**Comment by jguipi** [comment]
@ijjk my mistake I wanted to say Dynamic routing and not Dynamic API Routing. 
https://nextjs.org/docs/routing/dynamic-routes

**Comment by timneutkens** [comment, marked as answer]
Do you have a `pages/test/index.js` / `pages/test.js` or similar? This message means you are not exporting a React component

**Reply by jguipi** [reply]
That made it work thank
